pkg/services/ngalert/ngalert.go

```diff
@@ -268,6 +269,10 @@ func (ng *AlertNG) init() error {
 					cfg.OrgID = orgID
 					remoteAM, err := createRemoteAlertmanager(ctx, cfg, ng.KVStore, crypto, autogenFn, m, ng.tracer)
 					if err != nil {
+						if remoteSecondaryWithRemoteState {
```

Comment:
I was hoping this whole factory would move into the notify package.  It would be great to refactor it further

Reply:
I can do it in a separate PR. I don't think it should fall within the scope of this one.

Reply:
WIP -> https://github.com/grafana/grafana/pull/108582